hdate/tekufot.py

```diff
@@ -0,0 +1,245 @@
+            hdate_7_cheshvan = HebrewDate(self.hebrew_year_p, 2, 7)
+            jdn_7_cheshvan = conv.hdate_to_jdn(hdate_7_cheshvan)
+            cheilat_geshamim = conv.jdn_to_gdate(jdn_7_cheshvan)
+
+        return cheilat_geshamim
+
+    def get_cheilat_geshamim_hdate(self) -> HebrewDate:
+        """
+        Convert Cheilat Geshamim in Hebrew date.
+        """
+        jdn_geshamin = conv.gdate_to_jdn(self.get_cheilat_geshamim())
+
+        return conv.jdn_to_hdate(jdn_geshamin)
+
+    def get_gevurot(self) -> Gevurot:
+        """
+        From Pesach to Shemini Atzeret:
+          Israel & Sephardi: Morid (0)
+          Ashkenazi: neither (2)
+        From Shemini Atzeret to Next Pesach:
+          All: Mashiv (1)
+        """
+        # Prev Pesach to Shemini Atzeret
+        if (
+            HebrewDate(self.hebrew_year_p - 1, 7, 15)
+            < self.hebrew_date
+            < HebrewDate(self.hebrew_year_p, 1, 22)
+        ):
+            if self.tradition in ["israel", "diaspora_sephardi"]:
+                return Gevurot.MORID_HATAL  # Morid = 0
+
+            # diaspora_ashkenazi
+            return Gevurot.NEITHER  # neither = 2
+
+        # Shemini Atzeret to Pesach
+        if (
+            HebrewDate(self.hebrew_year_p, 1, 22)
+            < self.hebrew_date
+            < HebrewDate(self.hebrew_year_p, 7, 15)
+        ):
+            return Gevurot.MASHIV_HARUACH  # mashiv = 1
+
+        # Pesach to Next Shemini Atzeret
+        if (
+            HebrewDate(self.hebrew_year_p, 7, 15)
+            < self.hebrew_date
+            < HebrewDate(self.hebrew_year_p + 1, 1, 22)
+        ):
+            if self.tradition in ["israel", "diaspora_sephardi"]:
+                return Gevurot.MORID_HATAL  # Morid = 0
+
+            # diaspora_ashkenazi
+            return Gevurot.NEITHER  # neither = 2
+
+        # Par défaut (si rien ne correspond, ex. date hors de la plage) :
+        return Gevurot.NEITHER
+
+    def get_geshamim(self) -> Geshamim:
+        """
+        Periods:
+        From Pesach I (Musaf) to Cheilat geshamim: All = barkheinu (0)
+        Cheilat geshamim to Pesach I (Shacharit): All = barech_aleinu (1)
+        At Pesach I (Shacharit): All = barkheinu (0)
+        """
+
+        # From Prev Pesach to Cheilat geshamim: All barkheinu (0)
+        if (
+            HebrewDate(self.hebrew_year_p - 1, 7, 15)
+            < self.hebrew_date
+            < self.get_cheilat_geshamim_hdate()
+        ):
+            return Geshamim.BARKHEINU  # barkheinu = 0
+
+        # From Cheilat geshamim to Pesach: All = barech_aleinu (1)
+        if (
+            self.get_cheilat_geshamim_hdate()
+            < self.hebrew_date
+            < HebrewDate(self.hebrew_year_p, 7, 15)
+        ):
+            return Geshamim.BARECH_ALEINU  # barech_aleinu = 1
+
+        # At Pesach (Shacharit): All barkheinu (0)
+        if self.hebrew_date == HebrewDate(self.hebrew_year_p, 7, 15):
+            return Geshamim.BARKHEINU
```

Comment:
This is never tested, but does it make sense? On pesach you don't say neither Barcheinu nor barech aleinu, it's tfilat Yom tov.